commit 591209c79844c1ecbee79e6b5a019e5b61eab8d3
Author: Alison Schofield <alison.schofield@intel.com>
Date:   Sat Jul 6 18:53:43 2024 -0700

    cxl/memdev: Replace ENXIO with EBUSY for inject poison limit reached
    
    The CXL driver provides a debugfs interface offering users the
    ability to inject and clear poison to a memdev. Once a user has
    injected up to the devices limit further injection requests fail
    with ENXIO until a clear poison is issued.
    
    Users may not have device specs in hand or may want to intentionally
    hit the limit and then clear. Replace the usual ENXIO return status
    with EBUSY so users can recognize this failure.
    
    Signed-off-by: Alison Schofield <alison.schofield@intel.com>
    Tested-by: Xingtao Yao <yaoxt.fnst@fujitsu.com>
    Reviewed-by: Dan Williams <dan.j.williams@intel.com>
    Reviewed-by: Davidlohr Bueso <dave@stgolabs.net>
    Link: https://patch.msgid.link/825bd4c67fb55a4373c4182d999ad49d4e6b4fe7.1720316188.git.alison.schofield@intel.com
    Signed-off-by: Dave Jiang <dave.jiang@intel.com>

diff --git a/drivers/cxl/cxlmem.h b/drivers/cxl/cxlmem.h
index f55141c6d64e..40a465c37e03 100644
--- a/drivers/cxl/cxlmem.h
+++ b/drivers/cxl/cxlmem.h
@@ -161,7 +161,7 @@ struct cxl_mbox_cmd {
 	C(FWRESET, -ENXIO, "FW failed to activate, needs cold reset"),		\
 	C(HANDLE, -ENXIO, "one or more Event Record Handles were invalid"),     \
 	C(PADDR, -EFAULT, "physical address specified is invalid"),		\
-	C(POISONLMT, -ENXIO, "poison injection limit has been reached"),        \
+	C(POISONLMT, -EBUSY, "poison injection limit has been reached"),        \
 	C(MEDIAFAILURE, -ENXIO, "permanent issue with the media"),		\
 	C(ABORT, -ENXIO, "background cmd was aborted by device"),               \
 	C(SECURITY, -ENXIO, "not valid in the current security state"),         \